tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPMirrorConnectionTest.java

```diff
@@ -229,6 +271,173 @@ public void testBrokerAddsAddressAndQueue() throws Exception {
+         Wait.assertTrue(() -> server.locateQueue(getQueueName()) != null);
+
+         final org.apache.activemq.artemis.core.server.Queue q1 = server.locateQueue(getQueueName());
+         assertNotNull(q1);
+
+         final org.apache.activemq.artemis.core.server.Queue q2 = server_2.locateQueue(getQueueName());
+         assertNotNull(q2);
+
+         final ConnectionFactory factory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:" + BROKER_PORT_NUM);
+         final ConnectionFactory factory_2 = CFUtil.createConnectionFactory("CORE", "tcp://localhost:" + AMQP_PORT_2);
+
+         String message = "A message!";
+         if (longMessage) {
+            String sLongMessage = "";
+            for (int i = 0; i < 11 * 1014; i++) {
+               sLongMessage += message;
+            }
+            message = sLongMessage;
+         }
+
+         try (Connection conn = factory.createConnection()) {
+            final Session session = conn.createSession();
+            conn.start();
+
+            final MessageProducer producer = session.createProducer(session.createQueue(getQueueName()));
+            producer.send(session.createTextMessage(message));
+            producer.close();
+         }
+
+         org.apache.activemq.artemis.utils.Wait.assertEquals(1L, q1::getMessageCount, 100, 100);
+         org.apache.activemq.artemis.utils.Wait.assertEquals(1L, q2::getMessageCount, 100, 100);
+
+         try (Connection conn = factory_2.createConnection()) {
+            final Session session = conn.createSession();
+            conn.start();
+
+            final MessageConsumer consumer = session.createConsumer(session.createQueue(getQueueName()));
+            TextMessage rcvMsg = (TextMessage) consumer.receive(1000);
+            assertNotNull(message);
+            assertEquals(message, rcvMsg.getText());
+            consumer.close();
+         }
+
+         org.apache.activemq.artemis.utils.Wait.assertEquals(0L, q2::getMessageCount, 100, 100);
+         org.apache.activemq.artemis.utils.Wait.assertEquals(0L, q1::getMessageCount, 6000, 100);
+
+         // give some time to peer_3 to receive messages (if any)
+         Thread.sleep(100);
+         peer_3.waitForScriptToComplete(5, TimeUnit.SECONDS); // if messages are received here, this should error out
+
+         // Then send messages on the broker directly connected to the peer, the messages should make it to the peer.
+         // Receiving these 3 messages in that order confirms that no previous data reched the Peer, therefore validating
+         // the test.
+         peer_3.expectTransfer().accept(); // Address create
+         peer_3.expectTransfer().accept(); // Queue create
+         peer_3.expectTransfer().withMessageFormat(messageFormat).accept(); // Producer Message
+
+         server_2.addAddressInfo(new AddressInfo(SimpleString.of("sometest"), RoutingType.ANYCAST));
+         server_2.createQueue(QueueConfiguration.of("sometest").setRoutingType(RoutingType.ANYCAST));
```

Comment:
could probably use something more descriptive than "sometest". Either way, a variable would probably make it easier to follow where it is used, e.g in the mirror config for filtering, and again here/below.